lib/xcodeproj/project.rb

```diff
@@ -809,7 +809,7 @@ def recreate_user_schemes(visible = true)
       targets.each do |target|
         scheme = XCScheme.new
         scheme.add_build_target(target)
-        scheme.add_test_target(target) if target.test_target_type?
+        scheme.add_test_target(target) if target.instance_of?(Object::PBXNativeTarget) && target.test_target_type?
```

Comment:
we might want to implement this `if target.responds_to?(:test_target_type?) && target.test_target_type?`

Also needs a test

Reply:
I just really gave you an idea on what's the problem and a possible solution. I solved this issue on my own, by processing the targets exactly the same way proposed in here. Until a new version be released with this bug fixed. This is my first experience with Ruby ever, so can't tell much more.

Thanks a lot for the support, and for leting me contribute with this project.

Reply:
Tested responds_to method as follows:

![image](https://user-images.githubusercontent.com/6015754/34258880-c1115a60-e63e-11e7-8bd9-7ec3aabe5621.png)

Reply:
yes sorry please use `respond_to?` instead

Reply:
Ready